src/extension.ts

```diff
@@ -87,17 +87,18 @@ export function deactivate(): Thenable<void> | undefined {
 async function installThenStart(context: vscode.ExtensionContext, config: vscode.WorkspaceConfiguration) {
 	const command: string = config.get("languageServer.pathToBinary");
 	if (command) { // Skip install/upgrade if user has set custom binary path
-		startLsClient(command, config);
+		await startLsClient(command, config);
 	} else {
 		const installer = new LanguageServerInstaller;
-		const installDir = `${context.extensionPath}/lsp`
-		installer.install(installDir).then(() => {
-			config.update("languageServer.external", true, vscode.ConfigurationTarget.Global);
```

Comment:
@paultyng this change is a bit outside my current understanding of the codebase, just want to double check with you this is what we want?

Reply:
we should move this to the top level commands for enable/disable

Reply:
It appears to already be there, could I get approval? We tested this PR together already.